s3/s3_transport/include/s3_transport.hpp

```diff
@@ -1081,13 +1081,14 @@ namespace irods::experimental::io::s3_transport
 
                 initiate_multipart_upload();
 
-                irods::thread_pool cache_flush_threads{static_cast<int>(config_.number_of_cache_transfer_threads)};
-
                 unsigned int part_number = 1;
 
                 std::int64_t part_size_all_but_last_part = cache_file_size / number_of_parts;
+
                 while (part_number <= number_of_parts) {
 
+                    irods::thread_pool cache_flush_threads{static_cast<int>(config_.number_of_cache_transfer_threads)};
+
```

Comment:
This seems costly.

How often do you expect a new thread pool to be instantiated?

Reply:
Rarely.  For example, with the defaults of 10 threads and 5 GB maximum part size, for a 100 GB file it would get re-instantiated one time (2 total instantiations).  Whatever time it takes to re-instantiate it would only be a fraction compared to the time to actually send the bytes.

Reply:
I feel much better about the change now.

Please mark this conversation as resolved.